sycl/include/sycl/ext/oneapi/experimental/annotated_ptr/annotated_ptr.hpp

```diff
@@ -91,15 +62,44 @@ class annotated_ref<T, detail::properties_t<Props...>> {
   T *m_Ptr;
   explicit annotated_ref(T *Ptr) : m_Ptr(Ptr) {}
 
+  // properties filter
+  template <typename property_list, template <class...> typename filter>
+  using PropertiesFilter =
+      sycl::detail::boost::mp11::mp_copy_if<property_list, filter>;
```

Comment:
```suggestion
      detail::boost::mp11::mp_copy_if<property_list, filter>;
```

Reply:
This has to be kept when I test it

Reply:
why?

Reply:
Removing "sycl::" gets error:
```
/tmp/wangdi4/frontend/w3/llvm/build/include/sycl/ext/oneapi/experimental/annotated_ptr/annotated_ptr.hpp:68:15: error: ‘boost’ in namespace ‘sycl::_V1::ext::oneapi::experimental::detail’ does not name a type
   68 |       detail::boost::mp11::mp_copy_if<property_list, filter>;
```

Since the code above is written in `sycl::ext::oneapi::experimental` namespace, it cannot see the `detail` namespace, because `detail` namespace doesn't enclose it